orchestrator/graphql/schema.py

```diff
@@ -150,12 +147,33 @@ async def get_context(custom_context=Depends(custom_context_dependency)) -> Orch
     return custom_context
 
 
-def create_graphql_router(query: Any = Query, mutation: Any = Mutation) -> OrchestratorGraphqlRouter:
+GRAPHQL_MODELS_INITIAL = dict(GRAPHQL_MODELS)
+
+
+def create_graphql_router(
+    query: Any = Query,
+    mutation: Any = Mutation,
+    register_models: bool = True,
+    subscription_interface: Any = SubscriptionInterface,
+) -> OrchestratorGraphqlRouter:
+    @strawberry.experimental.pydantic.type(
+        model=SubscriptionModel, all_fields=True, directives=federation_key_directives
+    )
+    class Subscription(SubscriptionInterface):
+        pass
+
+    models = GRAPHQL_MODELS_INITIAL
+    models["subscription"] = Subscription
+
+    if register_models:
+        models = register_domain_models(subscription_interface, models)
+    GRAPHQL_MODELS.update(models)
```

Comment:
The intermediate `models` variable doesn't add much imho. Also the `register_domain_models` function modifies the dictionary itself, this can become a problem sometime. We could simplify and make it more explicit;

```suggestion
    GRAPHQL_MODELS["subscription"] = Subscription

    if register_models:
        register_domain_models(subscription_interface, models=GRAPHQL_MODELS)
```

Also, `register_domain_models` will modify `GRAPHQL_MODELS` directly if the `models` kwarg is None, I think it shouldn't do that